zebra/zebra_vty.c

```diff
@@ -3486,6 +3514,40 @@ DEFUN (show_evpn_neigh_vni_vtep,
 	return CMD_SUCCESS;
 }
 
+DEFPY(show_evpn_local_mac, show_evpn_local_mac_cmd,
+      "show evpn local-mac IFNAME$if_name (1-4094)$vid [json$json]",
+      SHOW_STR
+      "EVPN\n"
+      "Local MAC addresses\n"
+      "Interface Name\n"
+      "VLAN ID\n" JSON_STR)
+{
+	struct vrf *vrf = NULL;
+	struct interface *ifp = NULL;
+	bool found = false;
+	bool uj = use_json(argc, argv);
+
+	if (!if_name || !vid) {
+		return CMD_WARNING;
+	}
+
+	RB_FOREACH (vrf, vrf_name_head, &vrfs_by_name) {
```

Comment:
I swear - no more of this in-the-clear: let's make a lib api that does this, instead of doing it raw like this all over zebra